File: src/HolzShots.Legacy/Drawing/Tools/Scale.vb
Imports HolzShots.Drawing.Tools.UI

Namespace Drawing.Tools
    Friend Class Scale
        Implements IDialogTool

        Public Sub ShowToolDialog(ByRef rawImage As Image, screenshot As Screenshot, parent As IWin32Window) Implements IDialogTool.ShowToolDialog

            Using s As New ScaleWindow(rawImage)

                If s.ShowDialog(parent) <> DialogResult.OK Then
                    Return
                End If

                Dim width = s.WidthBoxV
                Dim height = s.HeightBoxV

                Dim unit As ScaleWindow.ScaleUnit = s.CurrentScaleUnit

                Dim newWidth As Integer = CInt(width)
                Dim newHeight As Integer = CInt(height)

                ' This is wrong and may fail if there was no cursor position
                ' We ignore this for now, since we're not caring about legacy stuff
                Dim newCursorCoordinates As Point = screenshot.CursorPosition.OnImage

                Dim newCursorWidth As Integer = My.Resources.windowsCursorMedium.Width
                Dim newCursorHeight As Integer = My.Resources.windowsCursorMedium.Height

                If unit = ScaleWindow.ScaleUnit.Percent Then
                    newWidth = CInt(rawImage.Width * (width / 100))
                    newHeight = CInt(rawImage.Height * (height / 100))
                    newCursorCoordinates.X = CInt(newCursorCoordinates.X * (height / 100))
                    newCursorCoordinates.Y = CInt(newCursorCoordinates.Y * (width / 100))
                    newCursorWidth = CInt(newCursorWidth * (width / 100))
                    newCursorHeight = CInt(newCursorHeight * (height / 100))
                End If

                Dim newRawImage As New Bitmap(newWidth, newHeight)
                Using g As Graphics = Graphics.FromImage(newRawImage)
                    g.DrawImage(rawImage, 0, 0, newWidth, newHeight)
                End Using

                rawImage = newRawImage
            End Using
        End Sub
        Public Sub LoadInitialSettings() Implements IDialogTool.LoadInitialSettings
            ' Nothing to do here
        End Sub
        Public Sub PersistSettings() Implements IDialogTool.PersistSettings
            ' Nothing to do here
        End Sub

        Public Sub Dispose() Implements IDialogTool.Dispose
        End Sub
    End Class
End Namespace

File: src/HolzShots.LegacyNew/Drawing/Tools/Scale.cs
using HolzShots.Drawing.Tools.UI;

namespace HolzShots.Drawing.Tools;
public class Scale : IDialogTool
{
    public void ShowToolDialog(ref Image rawImage, Screenshot screenshot, IWin32Window? parent)
    {
        using var s = new ScaleWindow(rawImage);

        if (s.ShowDialog(parent) != DialogResult.OK)
            return;

        var width = s.WidthBoxV;
        var height = s.HeightBoxV;

        ScaleUnit unit = s.CurrentScaleUnit;

        int newWidth = (int)width;
        int newHeight = (int)height;

        // This is wrong and may fail if there was no cursor position
        // We ignore this for now, since we're not caring about legacy stuff
        var newCursorCoordinates = screenshot.CursorPosition.OnImage;


        if (unit == ScaleUnit.Percent)
        {
            newWidth = (int)(rawImage.Width * (width / 100));
            newHeight = (int)(rawImage.Height * (height / 100));
            newCursorCoordinates.X = (int)(newCursorCoordinates.X * (height / 100));
            newCursorCoordinates.Y = (int)(newCursorCoordinates.Y * (width / 100));
        }

        var newRawImage = new Bitmap(newWidth, newHeight);

        using var g = Graphics.FromImage(newRawImage);
        g.DrawImage(rawImage, 0, 0, newWidth, newHeight);

        rawImage = newRawImage;
    }

    public void LoadInitialSettings() { }
    public void PersistSettings() { }
    public void Dispose() { }
}
